infra/terraform/global/assets/create_etcd_ebs_snapshot/tag/tag.go

```diff
@@ -0,0 +1,67 @@
+package tag
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/ec2"
+)
+
+func FetchResourceTags(connection *ec2.EC2, resourceID *string) (getTagsOutput *ec2.DescribeTagsOutput) {
```

Comment:
I think this is cleaner but usually the pattern is to return both the result and a potential error, e.g. 

```go
func FetchResourceTags(connection *ec2.EC2, resourceID *string) (getTagsOutput *ec2.DescribeTagsOutput, error) {
  //
  if (err) {
    return nil, err
  }

  // all good
  return snapshot, nil
}
```
(or something like this)

A part from being "Go idiomatic" which we may or may not care much, it transfer the responsibility of handling the error to the caller which has more context on what's the most appropriate course of action.

In our specific cases, when you return in this function it doesn't stop the snapshot loop, also the caller doesn't know if there was an error assuming there wasn't, e.g. it operate on the returned `snapshot` assuming this returned one.

Reply:
Yep, I totally agree.  In fact I started doing this in my prune snapshots function.  I'll get this refactored.  Thanks for the review.